File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int low = 0;
        int high = heights.length-1;
        int maxArea = 0;
        while(low<high){
            maxArea = Math.max(maxArea, Math.min(heights[low],heights[high])*(high-low));
            if(heights[low]<heights[high]){
                low++;
            }
            else{
                high--;
            }
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/coin-change/submission-0.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        if(amount == 0){
            return 0;
        }
        if(coins.size() == 1){
            if(amount%coins[0] == 0){
                return amount/coins[0];
            }
        }
        std::vector<int> dp(amount+1,10001);
        dp[0] = 0;
        for(int i = 0;i<coins.size();++i){
            for(int j = coins[i];j<dp.size();++j){
                dp[j] = std::min(dp[j], 1+dp[j-coins[i]]);
            }
            for(int i: dp){
                std::cout<<i<<",";
            }
        }
        if(dp[amount] == 10001){
            return -1;
        }
        return dp[amount];
    }
};
